cfme/common/vm.py

```diff
@@ -579,8 +579,7 @@ def create(self, vm_name, provider, form_values=None, cancel=False, check_existi
         if check_existing and vm.exists:
             return vm
         if not provider.is_refreshed():
-            provider.refresh_provider_relationships()
-            wait_for(provider.is_refreshed, func_kwargs={'refresh_delta': 10}, timeout=600)
+            provider.refresh_provider_relationships_and_wait()
```

Comment:
Nice PR ! One comment , Can we use this at line 414 as well ?

       def wait_to_appear(self, timeout=600, load_details=True):
        """Wait for a VM to appear within CFME
        Args:
            timeout: time (in seconds) to wait for it to appear
            load_details: when found, should it load the vm details
        """
        def _refresh():
            self.provider.refresh_provider_relationships()
            self.appliance.browser.widgetastic.browser.refresh()  # strange because ViaUI

        wait_for(
            lambda: self.exists,
            num_sec=timeout, delay=5, fail_func=_refresh,
            message="wait for vm to appear")
        if load_details:
            navigate_to(self, "Details", use_resetter=False)

Reply:
I added `_and_wait` and swapped it with widgetastic browser refresh. I think it makes sense like that.